### ISSUE TITLE
Same code works on iOS but not on Android

### ISSUE THREAD

**Issue by PrimeObjects** [aws/aws-sdk-js#1722]
Hello, 

Below is part of the code we used for testing the multi-part upload. It works fine on iOS but we got "The request signature we calculated does not match the signature you provided. Check your key and signing method." on android, any idea? 

Thanks
Gary

import AWS from 'aws-sdk/dist/aws-sdk-react-native';

AWS.config.update({accessKeyId: awsId, secretAccessKey: awsKey, region: 'us-west-2'});
        var s3 = new AWS.S3();
        s3.createMultipartUpload({ Bucket: bucket,  Key: 'abc.jpg', ContentType: 'image/jpeg', ContentEncoding: 'base64'},
        (err, data) => {
            if (err)
            {
                console.log(["failed",err, err.stack]);
            }
            else {
    console.log(["passed",data]);
            }
        })

**Comment by chrisradek** [comment]
@PrimeObjects 
Have you tried using `s3.upload`? For files larger than 5 MB, it will automatically handle mutli-part uploads. I was able to successfully upload a 10 MB image using iOS and android on react native version 0.47. What version of react native are you on? Can you try updating to at least version 0.47 if you aren't already at that version?

**Comment by PrimeObjects** [comment]
Hi Chrisradek,

Thank you for the help! It was actually the config ContentEncoding: 'base64' caused the problem. I simply removed it and it works fine now on Android and iOS.

Thanks
Gary

**Comment by jeskew** [comment]
@PrimeObjects Can I ask where you picked up the `Content-Encoding: 'base64'` approach from? I've seen that used in a number of React Native-related issues, but my understanding is that [it's disallowed by the HTTP spec](https://github.com/aws/aws-sdk-js/issues/1595#issuecomment-310855925). Is it used in official RN examples or a popular tutorial?

**Comment by PrimeObjects** [comment]
Hi jeskew, 

You know what I don't know why it was there at the first place. My team is new to React Native. I believe it was added by our React Native developer (he only has his hands dirty on RN for one week) who worked on file upload function. It was escalated up to me while he found the code worked on iOS but not on Android. I knew little about React Native, I tried to play around the configs and it worked after I removed this Content-Encoding. Simply luck. It took us time to find the issue, nobody questioned about config because it worked on iOS.

Gary
